library/X509/ExpirationWidget.php

```diff
@@ -66,21 +75,13 @@ protected function assemble()
             $state = 'state-ok';
         }
 
-        $this->add([
-            Html::tag(
-                'span',
-                ['class' => '', 'style' => 'font-size: 0.9em;', 'title' => $dateTip],
-                $message
-            ),
-            Html::tag(
-                'div',
-                ['class' => 'progress-bar dont-print'],
-                Html::tag(
-                    'div',
-                    ['style' => sprintf('width: %.2F%%;', $ratio), 'class' => "bg-stateful {$state}"],
-                    new HtmlString('&nbsp;')
-                )
-            )
-        ]);
+        $progressBar = Html::tag('div', ['class' => "bg-stateful $state"], new HtmlString('&nbsp;'));
+        $progressBarStyle = (new Style())
+            ->setModule(Icinga::app()->getRequest()->getModuleName())
+            ->setNonce(Csp::getStyleNonce())
+            ->addFor($progressBar, ['width' => sprintf('%.2F%%;', $ratio)]);
```

Comment:
Semicolon at the end is not required. `LessRuleset::renderLess()` does the job.

Reply:
And please remove `.2` from sprintf(), we don't want to set a limit to the digits after the decimal point.